labscript_devices/PrawnBlaster/runviewer_parsers.py

```diff
@@ -71,7 +71,19 @@ def get_traces(self, add_trace, clock=None):
 
         # Generate clocklines and triggers
         clocklines_and_triggers = {}
-        for pulse_program in pulse_programs:
+        
+        for pseudoclock_name, pseudoclock in self.device.child_list.items():
+            # Get pseudoclock index
+            connection_parts = pseudoclock.parent_port.split()
```

Comment:
Not really critical, but I'm reasonably certain you meant `connection_ports` instead of `connection_parts`, unless you were aiming for a more phonetic spelling ;)

Reply:
I did actually mean `connection_parts`. The `parent_port` is also known as the `connection` in other places. In fact, when I wrote the code I assumed it was `pseudoclock.connection` not `pseudoclock.parent_port` :shrug: So that `parts` suffix was due to my intention of calling `split()` (on an attribute called `connection`...and yes, I'm bad at variable naming!!). It should probably be `parent_port_parts` but I don't think I can be bothered changing it 😄